### PR TITLE
Install rspec and shoulda matchers

### PR DESCRIPTION
**Pull request by EevanR**
[PT](https://www.pivotaltracker.com/n/projects/2425705)

Installing rspec and shoulda matchers in appropriate files

### CODE REVIEW

**Comment by mantanquzzmo** [comment]
Fixed the merge conflicts

File: config/application.rb [hunk 1]
```diff
@@ -13,23 +12,20 @@
...
-    # Don't generate system test files.
-    config.generators.system_tests = nil
+    config.generators do |generate|
+      generate.helper false
+      generate.assets false
+      generate.view_specs false
+      generate.helper_specs false
+      generate.routing_specs false
+      generate.controller_specs false
+      config.generators.system_tests = nil
```
Comment (mantanquzzmo) [inline config/application.rb]: Might have to be exchanged to - generate.system_tests = nil
or put outside of |do| execution

Comment (oliverochman) [reply config/application.rb]: please refactor to `generate.system_tests = false`

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+#
+# Given that it is always loaded, you are encouraged to keep this file as
+# light-weight as possible. Requiring heavyweight dependencies from this file
+# will add to the boot time of your test suite on EVERY test run, even for an
+# individual file that may not need all of that loaded. Instead, consider making
+# a separate helper file that requires the additional dependencies and performs
+# the additional setup, and require it from the spec files that actually need
+# it.
+#
+# See http://rubydoc.info/gems/rspec-core/RSpec/Core/Configuration
```
Comment (oliverochman) [inline spec/spec_helper.rb]: clean this up

File: spec/rails_helper.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+rescue ActiveRecord::PendingMigrationError => e
+  puts e.to_s.strip
+  exit 1
+end
+
+RSpec.configure do |config|
+  config.fixture_path = "#{::Rails.root}/spec/fixtures"
+  config.use_transactional_fixtures = true
+  config.infer_spec_type_from_file_location!
+  config.filter_rails_from_backtrace!
```
Comment (oliverochman) [inline spec/rails_helper.rb:17]: add this line in this block
`  config.include FactoryBot::Syntax::Methods`

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: config/application.rb (hunk 1 above)
Comment (EevanR) [reply config/application.rb]: done

File: spec/spec_helper.rb (hunk 2 above)
Comment (EevanR) [reply spec/spec_helper.rb]: cleaned

File: spec/rails_helper.rb (hunk 3 above)
Comment (EevanR) [reply spec/rails_helper.rb:17]: added

**Pull request merged**